lib/PnpPlugin.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+const path = require("path");
+
+module.exports = class PnpPlugin {
+	constructor(source, pnpApi, target) {
+		this.source = source;
+		this.pnpApi = pnpApi;
+		this.target = target;
+	}
+
+	apply(resolver) {
+		const target = resolver.ensureHook(this.target);
+		resolver
+			.getHook(this.source)
+			.tapAsync("PnpPlugin", (requestContext, resolveContext, callback) => {
+				let request = requestContext.request;
+				let issuer = requestContext.context.issuer;
+
+				// When using require.context, issuer seems to be false (cf https://github.com/webpack/webpack-dev-server/blob/d0725c98fb752d8c0b1e8c9067e526e22b5f5134/client-src/default/index.js#L94)
+				if (!issuer) issuer = `${requestContext.path}/`;
+
+				// We only support issuer when they're absolute paths. I'm not sure the opposite can ever happen, but better check here.
+				if (!path.isAbsolute(issuer))
+					throw new Error(
```

Comment:
You can't throw here. Use the `callback` with an Error. (or expand the try catch below)

Reply:
But as `requestContext.path` is always absolute you can omit that.